### PR TITLE
Firebase backend

### PR DESCRIPTION
**Pull request by simonsayz13**
FIrebase functions added:
- Writing to RTDB
- Reading from RTDB
- Read/Write locations to RTDB. I have locations as a separate table (might not needed because we were saying to use static set of locations)
- @R-Middleton added "Warrington" to the locations context list

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,11 +1,40 @@
-import React from 'react';
+import React, {useEffect, useState}from 'react';
 import Hero from './components/hero';
 import Sidebar from './components/sidebar';
 import Content from './components/content';
 import Navbar from './components/navbar';
 import { LocationContextProvider } from './contexts/locationContext';
+import { readOpportunities, writeOpportunities, readLocations } from './firebase/firebaseHandler'
```
Comment (gavinroderick) [inline src/App.js:7]: Can remove `writeOpportunities` as it's unused

File: src/App.js [hunk 2]
```diff
@@ -1,11 +1,40 @@
...
+
+  useEffect( () => {
+    const loadData = async () => {
+      const opportunitiesData = await readOpportunities()
+      const locationsData = await readLocations()
+      setOpportunities(opportunitiesData)
+      setLocations(locationsData)
+
+      // Testing write opportunities
+      // writeOpportunities("10/08/22", "11/08/22", "descrip1","Leeds","Opportunity 1",5)
```
Comment (gavinroderick) [inline src/App.js:23]:
```suggestion
 
```

File: src/App.js [hunk 3]
```diff
@@ -1,11 +1,40 @@
...
+  const [locations, setLocations] = useState([])
+
+  useEffect( () => {
+    const loadData = async () => {
+      const opportunitiesData = await readOpportunities()
+      const locationsData = await readLocations()
+      setOpportunities(opportunitiesData)
+      setLocations(locationsData)
+
+      // Testing write opportunities
```
Comment (gavinroderick) [inline src/App.js:22]:
```suggestion

```

File: src/App.js (hunk 1 above)
Comment (gavinroderick) [inline src/App.js]:
```suggestion
import { readOpportunities, readLocations } from './firebase/firebaseHandler'
```

File: src/App.js [hunk 4]
```diff
@@ -1,11 +1,40 @@
...
 import Hero from './components/hero';
 import Sidebar from './components/sidebar';
 import Content from './components/content';
 import Navbar from './components/navbar';
 import { LocationContextProvider } from './contexts/locationContext';
+import { readOpportunities, writeOpportunities, readLocations } from './firebase/firebaseHandler'
 
 function App() {
+""
+  const [opportunities, setOpportunities] = useState([])
```
Comment (gavinroderick) [inline src/App.js:12]:
```suggestion
  const [setOpportunities] = useState([])
```

File: src/App.js [hunk 5]
```diff
@@ -1,11 +1,40 @@
...
+      // Testing write opportunities
+      // writeOpportunities("10/08/22", "11/08/22", "descrip1","Leeds","Opportunity 1",5)
+    }
+    
+    loadData()
+
+  }, []);
+  
+  // testing read locations (list of objects)
+  // console.log(locations)
```
Comment (gavinroderick) [inline src/App.js:31]: Can remove these comments also

File: src/App.js [hunk 6]
```diff
@@ -1,11 +1,40 @@
...
+  
+  // testing printing out location as text
```
Comment (gavinroderick) [inline src/App.js:33]: This can be removed

**Review by gavinroderick — CHANGES_REQUESTED** [review]
Great work! Just a little tidying up, removing comments and test code, then good to go! 👍🏻

File: src/App.js [hunk 7]
```diff
@@ -1,11 +1,46 @@
...
 import Hero from './components/hero';
 import Sidebar from './components/sidebar';
 import Content from './components/content';
 import Navbar from './components/navbar';
 import { LocationContextProvider } from './contexts/locationContext';
+//import writeOpportunities when needed
+import { readOpportunities, readLocations } from './firebase/firebaseHandler'
 
 function App() {
+""
```
Comment (gavinroderick) [inline src/App.js:11]: I think this can also be removed?
```suggestion

```

File: src/App.js (hunk 1 above)
Comment (simonsayz13) [reply src/App.js]: Removed

File: src/App.js (hunk 5 above)
Comment (simonsayz13) [reply src/App.js]: Done

**Review by gavinroderick — APPROVED** [review]
lgtm! 

one note, for consistency across our log, would you be able to use [commitizen](https://commitizen-tools.github.io/commitizen/) (or similar) for commits in the future?

**Pull request merged**